test/test_datasets_download.py

```diff
@@ -249,7 +249,8 @@ def voc():
 
 
 def mnist():
-    return collect_download_configs(lambda: datasets.MNIST(ROOT, download=True), name="MNIST")
+    with unittest.mock.patch.object(datasets.MNIST, "mirrors", datasets.MNIST.mirrors[-1:]):
+        return collect_download_configs(lambda: datasets.MNIST(ROOT, download=True), name="MNIST")
```

Comment:
This is a hacky fix, but the test infrastructure is currently not able to handle mirrors. During test collection, our tests mock out `download_url` and record all calls to for each dataset. Afterwards we do head requests against the logged URLs to check if they are accessible. Since we can't know if a request returns an error during collection, the mirror URLs are never collected.

Maybe we should add a mirror functionality to `download_url` that handles this internally. If we had this, we could add a functionality like this to our tests that warns in case a mirror is down.